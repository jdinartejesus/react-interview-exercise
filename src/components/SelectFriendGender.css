:local(.friendGender) {
  background-color: white;
  height: 34px;
  border: 1px solid #ccc;
}

:local(.friendGenderSelect) {
  border: none;
  padding: 6px;
  width: 100%;
  background: white;
  height: 32px;
}
