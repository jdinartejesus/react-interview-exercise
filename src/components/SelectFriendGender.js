import React, { Component, PropTypes } from 'react';
import styles from './SelectFriendGender.css';

class SelectFriendGender extends Component {
  constructor(props) {
    super(props);

    this.handleChange = this.handleChange.bind(this);
  }

  handleChange (e) {
    this.props.onChangeGender(e.target.value)
  }

  render () {
    const { value } = this.props

    return (
      <div className={styles.friendGender}>
        <select className={styles.friendGenderSelect} onChange={this.handleChange} value={value}>
          <option value="male">Male</option>
          <option value="female">Female</option>
        </select>
      </div>
    )
  }
}

const { string } = PropTypes

SelectFriendGender.propTypes = {
  value: string.isRequired
}

export default SelectFriendGender
