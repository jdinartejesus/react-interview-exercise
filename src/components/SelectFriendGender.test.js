import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import SelectFriendGender from './SelectFriendGender';


describe('components/SelectFriendGender', () => {

  const defaultValue = 'female';
  const onChangeGender = (value) => {
    return value;
  };

  it('should match it\'s empty snapshot', () => {
    const tree = renderer.create(
      <SelectFriendGender onChangeGender={onChangeGender} value={defaultValue}  />
    );

    expect(tree).toMatchSnapshot();
  })

  it('should contain two options', () => {
    const component = shallow(<SelectFriendGender onChangeGender={onChangeGender} value={defaultValue} />);

    expect(component.find('option')).toHaveLength(2);
  })

  it('should select the option male', () => {
    const onChangeGender = jest.fn();
    const component = shallow(<SelectFriendGender onChangeGender={onChangeGender} value="female" />);

    component.find('select').simulate('change', { target: { value : 'male'} });
    expect(onChangeGender).toBeCalledWith('male');
  })
})

