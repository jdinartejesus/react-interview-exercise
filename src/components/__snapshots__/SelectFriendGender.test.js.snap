exports[`components/SelectFriendGender should match it's empty snapshot 1`] = `
<div
  className={undefined}>
  <select
    className={undefined}
    onChange={[Function]}
    value="female">
    <option
      value="male">
      Male
    </option>
    <option
      value="female">
      Female
    </option>
  </select>
</div>
`;
